tests/forte/data/data_pack_profiling_test.py

```diff
@@ -0,0 +1,448 @@
+# Copyright 2019 The Forte Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+profiling test for data pack: using entry (0.2.0) and using new methods (>0.3.0).
+"""
+
+import os
+import unittest
+import nltk
+
+__all__ = [
+    "performance_test",
+]
+
+from typing import Dict, Set
+
+from forte.common.configuration import Config
+from forte.common.resources import Resources
+
+from forte.data.readers import OntonotesReader, DirPackReader, StringReader
+from forte.processors.writers import PackNameJsonPackWriter
+
+from ft.onto.base_ontology import EntityMention, Token, Sentence
+from nltk import (  # type: ignore
+    pos_tag,
+    ne_chunk,
+    PunktSentenceTokenizer,
+    download,
+)
+
+from nltk.tokenize import sent_tokenize, word_tokenize
+
+from forte.processors.base import PackProcessor
+from forte.data.data_pack import DataPack
+
+from forte import Pipeline
+from nltk.tokenize.treebank import TreebankWordTokenizer
+
+# from fortex.spacy import SpacyProcessor
+
+
+class SentenceAndTokenProcessor(PackProcessor):
+    def __init__(self):
+        super().__init__()
+
+    def initialize(self, resources, configs):
+        super().initialize(resources, configs)
+
+    def process_tokens(self, sentences, input_pack: DataPack):
+        """Basic tokenization and post tagging of the sentence.
+        Args:
+            processors: List of processor names.
+            sentences: Generator object which yields sentences in document.
+            input_pack: input pack which needs to be modified.
+        Returns: A mapping from SpaCy token index to Forte Token.
+        """
+        tokens: [Token] = []
+
+        last_sentence_word_idx = 0
+        for s_idx, sentence in sentences:
+            Sentence(input_pack, s_idx, s_idx + len(sentence))
+
+            for word in sentence:
+                begin_pos_word = word.idx
+                end_pos_word = begin_pos_word + len(word.text)
+                token = Token(input_pack, begin_pos_word, end_pos_word)
+                tokens.append(token)
+
+        return tokens
+
+    def _process(self, input_pack: DataPack):
+        doc = input_pack.text
+
+        sentences = sent_tokenize(doc)
+
+        # tokens = process_tokens(sentences, input_pack)   # sentences, input_pack
+        tokens: [Token] = []
```

Comment:
what is this format

Reply:
Format fixed and another one currently not used (by passed) method is removed form the code.